src/main/java/com/streamr/client/StreamrClient.java

```diff
@@ -113,18 +121,28 @@ public void onMessage(String message) {
                 public void onClose(int code, String reason, boolean remote) {
                     log.info("Connection closed! Code: " + code + ", Reason: " + reason);
                     state = State.Disconnected;
-                    StreamrClient.this.onClose();
+                    if (!remote) {
+                        StreamrClient.this.onClose();
+                    }
                 }
 
                 @Override
                 public void onError(Exception ex) {
                     log.error(ex);
-
-                    if (state == State.Connecting) {
-                        errorWhileConnecting = ex;
+                    if (ex instanceof IOException) {
+                        try {
+                            Thread.sleep(options.getConnectionTimeoutMillis());
```

Comment:
Would be nice to log something here before sleeping. Like: `Disconnected. Attempting to reconnect in X seconds`.